frontend/packages/ux-editor/src/components/config/componentSpecificContent/Summary2/Override/Summary2OverrideDisplaytype.tsx

```diff
@@ -0,0 +1,51 @@
+import React, { type ReactElement } from 'react';
+import { StudioCard, StudioNativeSelect } from '@studio/components';
+import type {
+  Summary2OverrideConfig,
+  SummaryCustomTargetType,
+} from 'app-shared/types/ComponentSpecificConfig';
+import { type CustomConfigType, useCustomConfigType } from './hook/useCustomConfigType';
+import { useTranslation } from 'react-i18next';
+
+export type Summary2OverrideDisplaytypeProps = {
+  override: Summary2OverrideConfig;
+  onChange: (override: Summary2OverrideConfig) => void;
+};
+
+export const Summary2OverrideDisplaytype = ({
+  override,
+  onChange,
+}: Summary2OverrideDisplaytypeProps) => {
+  const { t } = useTranslation();
+
+  const displayType = override.displayType || 'string';
+  if (!override.displayType) onChange({ ...override, displayType: displayType });
+
+  const handleCustomTypeChange = (event: React.ChangeEvent<HTMLSelectElement>) => {
+    const newType = event.target.value as SummaryCustomTargetType;
+    const updatedCustomType = { displayType: newType, componentId: override.componentId };
+    onChange(updatedCustomType);
+  };
```

Comment:
Can we maybe create a function outside the component that does this mapping, for example in a utils file next to this file? 🤔 Then we can test that function better 😄 

Something like this: 
```tsx
const handleCustomTypeChange = (event: React.ChangeEvent<HTMLSelectElement>): void => {
    const newSelectedType = event.target.value as SummaryCustomTargetType;
    const summary2OverrideConfig = mapSelectedTypeToConfig(newSelectedType) 
    onChange(summary2OverrideConfig);
 };
```
And then we need the `mapSelectedTypeToConfig` function: 
```tsx
export const mapSelectedTypeToConfig = (
   newSelectedType: SummaryCustomTargetType,
   componentId: string
): Summary2OverrideConfig => {
   return { 
      displayType: newSelectedType,
      componentId
   }
}
```

Reply:
I wonder why we need to make a seperate mapping for types?🙂 does not this belong to the same component `Summary2OverrideDisplayType`?  because we just handle change event to the types in `StudioNativeSelect`.  if so which utils file you mean?

Reply:
fixed